tests/lax_numpy_test.py

```diff
@@ -853,6 +864,51 @@ def np_fun(x):
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
     self._CompileAndCheck(jnp_fun, args_maker)
 
+  @unittest.skipIf(numpy_version < (1, 20), "where parameter not supported in older numpy")
+  @parameterized.named_parameters(itertools.chain.from_iterable(
+    jtu.cases_from_list(
+      {"testcase_name": "{}_inshape={}_axis={}_keepdims={}_whereshape={}".format(
+        rec.test_name.capitalize(),
+        jtu.format_shape_dtype_string(shape, dtype), axis, keepdims,
+        jtu.format_shape_dtype_string(whereshape, bool)),
+        "rng_factory": rec.rng_factory, "shape": shape, "dtype": dtype,
+        "np_op": getattr(np, rec.name), "jnp_op": getattr(jnp, rec.name), "whereshape": whereshape,
+        "axis": axis, "keepdims": keepdims, "inexact": rec.inexact}
+      for shape in rec.shapes for dtype in rec.dtypes
+      for whereshape in _compatible_shapes(shape)
+      for axis in list(range(-len(shape), len(shape))) + [None]
+      for keepdims in [False, True])
+    for rec in JAX_REDUCER_WHERE_NO_INITIAL_RECORDS))
+  def testReducerWhere(self, np_op, jnp_op, rng_factory, shape, dtype, axis,
+                              keepdims, inexact, whereshape):
+    if (shape in [()] + scalar_shapes and
+            dtype in [jnp.int16, jnp.uint16] and
+            jnp_op in [jnp.min, jnp.max]):
+      self.skipTest("Known XLA failure; see https://github.com/google/jax/issues/4971.")
+    rng = rng_factory(self.rng())
+    is_bf16_nan_test = dtype == jnp.bfloat16 and rng_factory.__name__ == 'rand_some_nan'
+    # Do not pass where via args_maker as that is incompatible with _promote_like_jnp.
+    where = jtu.rand_bool(self.rng())(whereshape, np.bool_)
```

Comment:
Another effect of passing this outside args_maker is that it is treated as static – i.e. the test does not cover the case that `where` is a traced value. I see this comes from the existing `where` test, but it's a bit worrisome in terms of test coverage.

Reply:
So `args_maker = lambda: [rng(shape, dtype)]` should also contain `[..., rng(whereshape, np.bool_)]` and `jnp_fun = lambda x, where: ...`? Should I rewrite it?

Reply:
That sounds right, though I think it might make the type promotion wrapper not work correctly, so we'd have to use `check_dtypes=False` in the numpy check.

I could go either way on this - it's up to you.